src/test/regress/expected/with_modifying.out

```diff
@@ -964,6 +964,15 @@ DETAIL:  distribution column value: 1
 (1 row)
 
 RESET client_min_messages;
+-- https://github.com/citusdata/citus/issues/3975
+WITH mb AS (UPDATE modify_table SET val = 3 WHERE id = 3 RETURNING NULL) SELECT * FROM modify_table WHERE id = 3;
+ id | val
+---------------------------------------------------------------------
+(0 rows)
+
+WITH mb AS (UPDATE modify_table SET val = 3 WHERE id = 3 RETURNING NULL) SELECT * FROM modify_table, mb WHERE id = 3;
```

Comment:
With non-distributed tables this results in 3 columns, the last being unnamed `?column?`

Reply:
We seem to have a general issue with unnamed columns:
```sql
WITH a AS (SELECT '' FROM test) SELECT * FROM a;
NOTICE:  issuing SELECT worker_column_1 AS "?column?" FROM (SELECT a."?column?" AS worker_column_1 FROM (SELECT ''::text FROM public.test_102009 test) a) worker_subquery
DETAIL:  on server marco@localhost:9701 connectionId: 6
```

Reply:
That's actually a regression from 9.2, I opened an issue: https://github.com/citusdata/citus/issues/3985

though the regression was introduced by inlining, so the obvious fix (not inlining) might not help the modifying CTE case.

Reply:
I think the proper fix would be to explicitly name the columns, e.g. this works:
```sql
WITH a AS (UPDATE public.test SET y = 3 WHERE x = 1 RETURNING NULL) SELECT "?column?" FROM a AS a ("?column?");
```

Reply:
Actually, it would be better to explicitly name the columns in the CTE.

```
WITH a ("?column?") AS (UPDATE public.test SET y = 3 WHERE x = 1 RETURNING NULL) SELECT "?column?" FROM a;
```

That way, we can do this for modifying CTEs in router queries without looking at the rest of the query tree.